src/rules/verify-user-email-with-password-reset.js

```diff
@@ -26,13 +26,17 @@ function (user, context, callback) {
       },
       json: { email_verified: true },
       timeout: 5000
-    }, 
+    },
     function(err, response, body) {
-      // Setting email verified isn't propaged to id_token in this 
-      // authentication cycle so explicitly set it to true 
-      context.idToken.email_verified = true;
+      // Setting email verified isn't propagated to id_token in this
+      // authentication cycle so explicitly set it to true given no errors.
+      if (err) {
```

Comment:
we should probably check `response.statusCode` here as well

is this too much "magic"
```
context.idToken.email_verified = (!err && response.statusCode === 200);
```

Reply:
Very nice :+1: